### ISSUE TITLE
ASAN Error while deleting a key in table of table of vector

### ISSUE THREAD

**Issue by msaboor35** [zeek/zeek#1931]
I am using the master version to reproduce the crash.

```
event zeek_init() {
        # define a table of table of vector of count
        local my_var: table[string] of table[string] of vector of count;

	#initialize table with a key which as 2 keys and empty vectors.
	my_var["a"] = table(["1"]=vector(), ["2"]=vector());

	# append an entry in vector
	my_var["a"]["1"] += 16;

	# detete the key
	delete my_var["a"];
}
```
I get the crash only when I try to delete a key in the table (inside which I appended some entry in vector). I will not get a crash if I ```delete my_var["a"]["2"]``` in the above POC as I don't append any entry in the vector in that key. Furthermore, keys can be string or count, we still get the crash. Instead of appending we can also use ```my_var["a"]["1"][0] = 16;``` to reproduce the crash. The crash is not reproduced in ```table[string] of vector of count```. 

Here is the crash:
```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==64288==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000010 (pc 0x559b1e76aeac bp 0x7ffc0ce5d7b0 sp 0x7ffc0ce5d680 T0)
==64288==The signal is caused by a READ memory access.
==64288==Hint: address points to the zero page.
    #0 0x559b1e76aeab in zeek::Unref(zeek::Obj*) /home/saboor/zeek/src/zeek/zeek/zeek/zeek/Obj.h:186
    #1 0x559b1e76aeab in zeek::ZVal::DeleteManagedType(zeek::ZVal&) /home/saboor/zeek/src/zeek/zeek/ZVal.h:146
    #2 0x559b1e76aeab in zeek::VectorVal::~VectorVal() /home/saboor/zeek/src/Val.cc:3213
    #3 0x559b1e76b272 in zeek::VectorVal::~VectorVal() /home/saboor/zeek/src/Val.cc:3217
    #4 0x559b1e768c72 in zeek::Unref(zeek::Obj*) /home/saboor/zeek/src/zeek/zeek/zeek/zeek/Obj.h:190
    #5 0x559b1e768c72 in zeek::IntrusivePtr<zeek::Val>::~IntrusivePtr() /home/saboor/zeek/src/zeek/zeek/IntrusivePtr.h:114
    #6 0x559b1e768c72 in zeek::TableEntryVal::~TableEntryVal() /home/saboor/zeek/src/zeek/Val.h:679
    #7 0x559b1e768c72 in table_entry_val_delete_func /home/saboor/zeek/src/Val.cc:1361
    #8 0x559b1dec3787 in zeek::Dictionary::Clear() /home/saboor/zeek/src/Dict.cc:979
    #9 0x559b1dec3eeb in zeek::Dictionary::~Dictionary() /home/saboor/zeek/src/Dict.cc:967
    #10 0x559b1e76b709 in zeek::PDict<zeek::TableEntryVal>::~PDict() /home/saboor/zeek/src/zeek/zeek/Dict.h:545
    #11 0x559b1e76b709 in zeek::TableVal::~TableVal() /home/saboor/zeek/src/Val.cc:1450
    #12 0x559b1e76c872 in zeek::TableVal::~TableVal() /home/saboor/zeek/src/Val.cc:1453
    #13 0x559b1d8d3ff3 in zeek::Unref(zeek::Obj*) /home/saboor/zeek/src/zeek/zeek/zeek/Obj.h:190
    #14 0x559b1e0255b5 in zeek::IntrusivePtr<zeek::Val>::~IntrusivePtr() /home/saboor/zeek/src/zeek/zeek/zeek/zeek/zeek/IntrusivePtr.h:114
    #15 0x559b1e0255b5 in zeek::detail::IndexExpr::Delete(zeek::detail::Frame*) /home/saboor/zeek/src/Expr.cc:2953
    #16 0x559b1e657fff in zeek::detail::DelStmt::Exec(zeek::detail::Frame*, zeek::detail::StmtFlowType&) /home/saboor/zeek/src/Stmt.cc:1067
    #17 0x559b1e660192 in zeek::detail::StmtList::Exec(zeek::detail::Frame*, zeek::detail::StmtFlowType&) /home/saboor/zeek/src/Stmt.cc:1629
    #18 0x559b1e660192 in zeek::detail::StmtList::Exec(zeek::detail::Frame*, zeek::detail::StmtFlowType&) /home/saboor/zeek/src/Stmt.cc:1629
    #19 0x559b1e2eeeee in zeek::detail::ScriptFunc::Invoke(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, zeek::detail::Frame*) const /home/saboor/zeek/src/Func.cc:435
    #20 0x559b1df64f31 in zeek::EventHandler::Call(std::vector<zeek::IntrusivePtr<zeek::Val>, std::allocator<zeek::IntrusivePtr<zeek::Val> > >*, bool) /home/saboor/zeek/src/EventHandler.cc:107
    #21 0x559b1df5c9b7 in zeek::Event::Dispatch(bool) /home/saboor/zeek/src/Event.cc:57
    #22 0x559b1df606ad in zeek::EventMgr::Drain() /home/saboor/zeek/src/Event.cc:158
    #23 0x559b1dd9ac13 in zeek::detail::setup(int, char**, zeek::Options*) /home/saboor/zeek/src/zeek-setup.cc:923
    #24 0x559b1d835962 in main /home/saboor/zeek/src/main.cc:13
    #25 0x7fb22d8bf0b2 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x270b2)
    #26 0x559b1d83573d in _start (/usr/local/zeek/bin/zeek+0x541073d)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/saboor/zeek/src/zeek/zeek/zeek/zeek/Obj.h:186 in zeek::Unref(zeek::Obj*)
==64288==ABORTING
```

**Comment by msaboor35** [comment]
I believe the problem arises because it mistakenly thinks count as a managed type while adding the entry in vector and when deleting it also tries to delete it as a managed type.

**Comment by vpax** [comment]
This is likely on me, I'll take a look.

**Comment by 0xxon** [comment]
Fixed in #1932 / becc966106aa2540755bf9805213f8feed8afb93
